operator/cmd/cilium_node.go

```diff
@@ -191,7 +209,16 @@ func startSynchronizingCiliumNodes(ctx context.Context, clientset k8sClient.Clie
 					return
 				}
 				if nodeManager != nil {
-					nodeManagerQueue.Add(key)
+					// Only enqueue key if already synced, handle at once while initializing
+					select {
+					case <-k8sCiliumNodesCacheSynced:
+						nodeManagerQueue.Add(key)
+					default:
+						err = nodeManagerSyncHandler(key)
+						if err != nil {
+							log.WithError(err).Errorf("initial sync %q failed with %v", key, err)
+						}
+					}
```

Comment:
We've got the exact same code in three places. Would it be worth refactoring a bit?